pkg/datagatherer/istio/istio_test.go

```diff
@@ -0,0 +1,166 @@
+	if err != nil {
+		t.Fatalf("unexpected error creating data gatherer: %+v", err)
+	}
+
+	// Fetch analysis result from the data gatherer.
+	rawAnalysisResult, err := dataGatherer.Fetch()
+	if err != nil {
+		t.Fatalf("unexpected error fetching results: %+v", err)
+	}
+
+	// Unpack the analysis result to find the code of the first message for checking.
+	var analysisResult map[string]interface{}
+	if err := json.Unmarshal([]byte(rawAnalysisResult.(string)), &analysisResult); err != nil {
+		t.Fatalf("unexpected error unmarshalling analysis result: %+v", err)
+	}
+	analysisMessages, ok := analysisResult["Messages"].([]interface{})
+	if !ok {
+		t.Fatalf("%+v", analysisResult["Messages"])
+	}
+	analysisMessage, ok := analysisMessages[0].(map[string]interface{})
+	if !ok {
+		t.Fatalf("%+v", analysisMessages[0])
+	}
+	analysisMessageCode, ok := analysisMessage["code"].(string)
+	if !ok {
+		t.Fatalf("%+v", analysisMessage["code"])
+	}
+
+	// With the test configuration and fake resources provided there should be only one message to warn about a missing
+	// Istio annotation on the default Namespace, with code IST0102.
+	if analysisMessageCode != "IST0102" {
+		t.Fatalf("unexpected analysis result messages: %+v", analysisMessageCode)
+	}
+}
+
+var configString = `
+kubeconfig: %s
+resources:
+- group:    ""
+  version:  "v1"
+  resource: "namespaces"
+`
+
+// createKubeConfigWithServer creates a kubeconfig file on disk with a reference to the local server.
+func createKubeConfigWithServer(server string) (string, error) {
+	content := fmt.Sprintf(kuebConfigString, server)
+	tempFile, err := ioutil.TempFile("", tempFilePrefix)
+	if err != nil {
+		return "", fmt.Errorf("failed to create a tmpfile for kubeconfig")
+	}
+
+	if _, err := tempFile.Write([]byte(content)); err != nil {
+		return "", fmt.Errorf("failed to write to tmp kubeconfig file")
+	}
+	if err := tempFile.Close(); err != nil {
+		return "", fmt.Errorf("failed to close tmp kubeconfig file after writing")
+	}
+
+	return tempFile.Name(), nil
+}
+
+var kuebConfigString = `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: %s
+  name: example
+contexts:
+- context:
+    cluster: example
+    namespace: default
+    user: test
+  name: test
+current-context: test
+users:
+- name: test
+  user:
+    username: test
+    password: test
+`
+
+// createLocalTestServer creates a local test server to respond to Kubernetes API requests from the dynamic data
+// gatherer.
+func createLocalTestServer(t *testing.T) *httptest.Server {
+	var localServer *httptest.Server
+	localServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var responseContent []byte
+
+		switch r.URL.Path {
+		case "/api/v1/namespaces":
+			responseContent = []byte(testNamespaces)
+		default:
+			t.Fatalf("Unexpected URL was called: %s", r.URL.Path)
+		}
+
+		w.Write(responseContent)
+	}))
+
+	return localServer
+}
+
+var testNamespaces = `
```

Comment:
optional: this is short enough that it can be inside the createlocaltestserver case statement.

Reply:
I'm inclined to leave it out, I think the function is easier to read without this string in and it will require less change if the testing is expanded in future.